checker/AnDO_engine.py

```diff
@@ -6,27 +6,114 @@
 import re
 import pathlib
 from AnDO_Error import (
+    ExperimentError,
     SourceError,
+    SourceNotFound,
     SessionError,
     SubError)
-
+from collections import OrderedDict
+import itertools
+import pprint
 dir_rules = os.path.join(os.path.dirname(__file__)) + '/rules/'
 
+def parse_all_path(nested_list_of_dir):
+    """
+    Transform this 
+    [
+        ['Landing', 'sub-anye', '180116_001_m_anye_land-001', 'source'],
+        ['Landing', 'sub-enya', '180116_001_m_enya_land-001', 'source'],
+        ['Landing', 'sub-enyo'], 
+        ['Landing', 'sub-enyo', '180116_001_m_enyo_land-001']
+    ]
+    to 
+    [
+        ['Landing', 'sub-anye', '180116_001_m_anye_land-001', 'source'],
+        ['Landing', 'sub-enya', '180116_001_m_enya_land-001', 'source'],
+    ]
+    Checking for the longest chain with the same sub chain
+    """
+    
+    main_list = sorted(nested_list_of_dir, key= lambda sublist: len(sublist))
+
+    # Méthode pour regrouper les sublist à utiliser sur la liste triée.
+    i=0
+    j=1
+    while i < len(main_list) -1:
+        if j <= len(main_list) -1:
+            if len(main_list[i]) <= len(main_list[j]):
+                all_in = True
+                for elmt in main_list[i]:
+                    if elmt not in main_list[j]:
+                        all_in = False
+
+                if all_in:
+                    for elmt_to_add in main_list[j]:
+                        if elmt_to_add not in main_list[i]:
+                            main_list[i].append(elmt_to_add)
+                    main_list.pop(j)
+
+                else:
+                    if j < len(main_list) - 1:
+                        j+=1
+                    else:
+                        i +=1
+                        j =i+1
+            else:
+                i+=1
+                j=i+1
+        else:
+            break
```

Comment:
this does not look like a very efficient algorithm to me, but we can work on this in the future ;)

Reply:
can you reply to the corresponding comments when you've done something to fix what was asked in the comment... even if it's juste one word like "done"... this will facilitate the reviewing! ;)